### PR TITLE
Update table header context menu 

### PR DESCRIPTION
**Pull request by julieg18**
* Adds "Update Header Depth" item to header 

## Demo

https://user-images.githubusercontent.com/43496356/193711009-db465cf0-ae59-4955-aadd-c6f14fb25eb2.mov

Related to #2390

### CODE REVIEW

File: extension/src/experiments/webview/messages.ts [hunk 1]
```diff
@@ -136,6 +136,9 @@ export class WebviewMessages {
           this.showPlots()
         ])
 
+      case MessageFromWebviewType.UPDATE_HEADER_DEPTH_SETTINGS:
+        return commands.executeCommand('workbench.action.openWorkspaceSettings')
```
Comment (julieg18) [inline extension/src/experiments/webview/messages.ts:140]: The closest thing I could find to linking to the dvc config settings was to just link to workspace settings in general... 

The only doable solution I can think of is creating a command that allows the user to update the config from the command palette 🤔

Comment (mattseddon) [reply extension/src/experiments/webview/messages.ts:140]: All you need to do is use [setConfigValue](https://github.com/iterative/vscode-dvc/blob/1efac498e4589de18877530c7210290de51a0254/extension/src/vscode/config.ts#L19). You can set the value through any means you want.

Comment (mattseddon) [reply extension/src/experiments/webview/messages.ts:140]: You will need to use a `quickInput` to get the required number.

File: webview/src/experiments/components/App.test.tsx [hunk 2]
```diff
@@ -702,6 +702,7 @@ describe('App', () => {
       const itemLabels = menuitems.map(item => item.textContent)
       expect(itemLabels).toStrictEqual([
         'Open to the Side',
+        'Update Header Depth',
```
Comment (mattseddon) [inline webview/src/experiments/components/App.test.tsx:705]: [I] Should be `Set Max Header Depth`.

**Review by mattseddon — APPROVED** [review]
Should consolidate the use of layers and depth. Would be good to see a test too.

**Review by sroy3 — APPROVED** [review]

**Comment by julieg18** [comment]
> Should consolidate the use of layers and depth. Would be good to see a test too.

Updated the text/variables, using depth instead of layers! And apologies, forgot tests again 😅 Will open them up in a followup!

File: extension/src/vscode/title.ts [hunk 3]
```diff
@@ -17,7 +17,8 @@ export enum Title {
   SELECT_PLOTS = 'Select Plots to Display',
   SELECT_SORT_DIRECTION = 'Select Sort Direction',
   SELECT_SORTS_TO_REMOVE = 'Select Sort(s) to Remove',
-  SETUP_WORKSPACE = 'Setup the Workspace'
+  SETUP_WORKSPACE = 'Setup the Workspace',
+  SET_EXPERIMENTS_HEADER_DEPTH = 'Set Maximum Amount of Experiment Table Depth'
```
Comment (mattseddon) [inline extension/src/vscode/title.ts:21]: Don't need `Amount of`.

Comment (mattseddon) [reply extension/src/vscode/title.ts:21]: Probably want to add `Header`

**Comment by shcheklein** [comment]
Thanks @julieg18 for the demo and the feature! Quick follow up questions / suggestions:

- Why there is a delay after applying the setting? Or at least it seems so ...
- In the quick pick menu let's add the description - e.g. that 0 means unlimited
- Minor: Depth probably is not the right term (should it be just height for simplicity?)

**Comment by julieg18** [comment]
> Why there is a delay after applying the setting? Or at least it seems so ...

Yes, there is a delay! It takes a sec for the table to rebuild. 

> In the quick pick menu let's add the description - e.g. that 0 means unlimited

Sure! I'll add it in a followup.

> Minor: Depth probably is not the right term (should it be just height for simplicity?)

"height" sounds good to me! Will update in a followup :)

**Comment by shcheklein** [comment]
Thanks!

> Yes, there is a delay. It takes a sec for the table to rebuild.

Can is be faster? Does it run a CLI command inside?

**Pull request merged**